### PR TITLE
[feature] Allow settings bot to modify env vars

### LINKED ISSUE #82: [feature] Mod channel designation
### Is your feature request related to a problem? Please describe

Epic: #26

As per conversation with kevin, we need will have one mod channel for all moderation-based message (modchat, salary TSA precheck)

### Describe the solution you'd like

command in settings to save channel

### Describe alternatives you've considered

N/A

### Additional context

- [ ] This may require a config file to save into rather than DB entry

### PR DESCRIPTION
**Pull request by sirNikolai**
### Purpose of pull request

**Related Issue**: resolves #82

Lets settings bot update environment variables and set mod communication channel

### Pull request checklist

- [x] Branch and PR are named correctly
- [x] Updated relevant documentation
- [x] Tested with a tester bot
- [x] Wrote unit tests for changes

### Testing instructions

`++settings update PREFIX --` - Updates environment variable (in this case the prefix)
`++settings setChannel` - updates mod comm channel to current one

### CODE REVIEW

File: src/dispatcher.js [hunk 1]
```diff
@@ -7,7 +7,9 @@ import client from './client';
 export const dispatchCommand = (message) => {
   client.message = message;
   if (message.content.startsWith(client.prefix)) {
-    const command = message.content.substring(2).split(' ')[0];
+    const command = message.content
+      .substring(client.prefix.length)
```
Comment (sirNikolai) [inline src/dispatcher.js:11]: the unit tests caught something 😄 

Case of if we ever set the prefix to something len 3+ need to adjust for that.

File: src/bots/settings/subCommands/updateEnvVar.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  if (!updateElement(args[0], args[1])) {
+    client.message.channel.send(
+      'Variable not found. Available options: ' + environmentElements.join(', ')
+    );
+    return;
+  }
+
+  client.message.channel.send(`Updated ${args[0]} to ${args[1]}`);
+  client.prefix = get('PREFIX');
```
Comment (sirNikolai) [inline src/bots/settings/subCommands/updateEnvVar.js:32]: I'm generally working off the premise that settings bot is the exception case and is the only bot that will be allowed modify any internal values in megabot

File: src/environment.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ */
+export const get = (arg) => {
+  return process.env[arg];
+};
+
+/**
+ * Rewrites .env file for updated values
+ */
+let updateEnvironment = () => {
+  //reference: https://stackoverflow.com/questions/55660763/how-to-generically-update-an-existing-environment-variable-to-env-file
```
Comment (brendacs) [inline src/environment.js:43]: Move this to the above JSdoc comment

File: src/dispatcher.js (hunk 1 above)
Comment (brendacs) [reply src/dispatcher.js:11]: eh yeah, I figured we never would really change it or have a need to make it longer

File: .env.template [hunk 4]
```diff
@@ -2,3 +2,5 @@ BOT_TOKEN=yourbottoken
 MONGODB=mongodb://localhost:27017/megabot
 ENV=testing
 GUILD_ID=1234567890
+MOD_CHANNEL_ID=0987654321
+PREFIX=++
```
Comment (brendacs) [inline .env.template:6]: newline

**Review by brendacs — COMMENTED** [review]
small comments, otherwise looks great

**Review by brendacs — APPROVED** [review]

**Pull request merged**